### PR TITLE
Create db migrations framework

### PR DESCRIPTION
**Pull request by phsmit**

### CODE REVIEW

File: models/migrations/migrations.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package migrations
+
+import (
+	"errors"
+	"github.com/go-xorm/xorm"
```
Comment (Unknwon) [inline models/migrations/migrations.go]: An empty line before this.

File: models/migrations/migrations.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+type migration func(*xorm.Engine) error
+
+// The version table. Should have only one row with id==1
+type Version struct {
+	Id      int64 `xorm:"pk"`
```
Comment (Unknwon) [inline models/migrations/migrations.go]: tag is verbose here

File: models/migrations/migrations.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+// Migrate database to current version
+func Migrate(x *xorm.Engine) error {
+	x.Sync(new(Version))
+
+	currentVersion := &Version{Id: 1}
+	has, err := x.Get(currentVersion)
+	if err != nil {
+		return err
+	}
+	if !has {
```
Comment (Unknwon) [inline models/migrations/migrations.go]: use `else if !has`

File: models/migrations/migrations.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		_, err = x.InsertOne(currentVersion)
```
Comment (Unknwon) [inline models/migrations/migrations.go]: didn't handle this error?

File: models/migrations/migrations.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	Version int64
+}
+
+// This is a sequence of migrations. Add new migrations to the bottom of the list.
+// If you want to "retire" a migration, replace it with "expiredMigration"
+var migrations = []migration{}
+
+// Migrate database to current version
+func Migrate(x *xorm.Engine) error {
+	x.Sync(new(Version))
```
Comment (Unknwon) [inline models/migrations/migrations.go]: Handle error here?

File: models/migrations/migrations.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	v := currentVersion.Version
+
+	for i, migration := range migrations[v:] {
+		if err = migration(x); err != nil {
+			return err
+		}
+		currentVersion.Version = v + int64(i) + 1
+		x.Id(1).Update(currentVersion)
```
Comment (Unknwon) [inline models/migrations/migrations.go]: handle error here, too

**Comment by phsmit** [comment]
All error should be catched now :)

**Comment by Unknwon** [comment]
Thanks! Merging...

**Pull request merged**